src/functions.js

```diff
@@ -88,6 +136,24 @@ function paintShop( cars, colour ){
 
   // the original array passed in should not change
   // hint: look up 'Cloning objects in JavaScript'
+    //var a = Object.assign({},cars[i])
+
+/*  doesnt work   var newCars=cars.slice();
+    console.log(newCars)
+  for (var i = 0; i < newCars.length; i++) {
+    for (var key in newCars[i]) {
+        if(newCars[i][key]==='Ford'){
+            newCars[i]['colour']=colour;
+    }}}
+    return newCars;
+  } */
+  var copy = [];
+  for (i = 0; i < cars.length; i++) {
+    copy[i] = Object.assign({}, cars[i]);
+    if (copy[i].make === 'Ford')
+      copy[i].colour = colour;
+  }
+  return copy;
 }
```

Comment:
This is an interesting approach to the problem. Definitely one where you could consider an alternative approach e.g. create copy array as first step, then loop through copy to amend its contents.